data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/EventHandle.java

```diff
@@ -14,4 +17,46 @@ public interface EventHandle {
      * @since 2.2
      */
     void release(boolean result);
+
+    /**
+     * sets acknowledgement set
+     *
+     * @param acknowledgementSet acknowledgementSet to be set in the event handle
+     * @since 2.6
+     */
+    void setAcknowledgementSet(final AcknowledgementSet acknowledgementSet);
+
+    /**
+     * gets acknowledgement set
+     *
+     * @return returns acknowledgementSet from the event handle
+     * @since 2.6
+     */
+    AcknowledgementSet getAcknowledgementSet();
+
+    /**
+     * sets external origination time
+     *
+     * @param externalOriginationTime externalOriginationTime to be set in the event handle
+     * @since 2.6
+     */
+    void setExternalOriginationTime(final Instant externalOriginationTime);
```

Comment:
I think we should set the origination time on the EventMetadata. The EventHandle can then get this data from the EventMetadata.

Reply:
Internal origination time is set in EventMetadata and EventHandle gets it from it. Do you want the external origination time also set in the EventMetadata?

Reply:
Yes, add this to `EventMetadata`.

The `EventHandle` can also hold a reference to the time in there.